packages/trader/src/Modules/CFD/Containers/compare-accounts-modal.tsx

```diff
@@ -13,18 +13,98 @@ import {
+) => Array<{ [key: string]: string | React.ReactNode | undefined }>;
+
+type TAccountsDescription = {
+    attribute: string;
+    mt5: TDxTradeAccountsDescription & { financial_stp: string };
+    dxtrade: TDxTradeAccountsDescription;
+};
+
+type TDxTradeAccountsDescription = {
+    synthetic: string;
+    synthetic_eu: string;
+    financial: string;
+    financial_au: string;
+    financial_eu: string | React.ReactNode;
+    footnote: string | null;
+};
+
+type TCompareAccountsReusedProps = {
+    landing_companies: LandingCompany;
+    platform: string;
+    is_logged_in: boolean;
+    is_uk: boolean;
+};
+
+type TCompareAccountsDataParams = TCompareAccountsReusedProps & {
+    show_eu_related: boolean;
+    residence: string;
+};
+
+type TCFDCompareAccountHintProps = TCompareAccountsReusedProps & {
+    show_risk_message: boolean;
+};
+
+type TModalContentProps = TCompareAccountsReusedProps & {
+    show_eu_related: boolean;
+    residence: string;
+    is_eu: boolean;
+};
+
+type TCompareAccountsModalProps = TCompareAccountsReusedProps & {
+    disableApp: () => void;
+    enableApp: () => void;
+    is_compare_accounts_visible: boolean;
+    is_loading: boolean;
+    is_eu: boolean;
+    is_eu_country: boolean;
+    residence: string;
+    toggleCompareAccounts: () => void;
+};
+
+type TGetAccounts = (params: TCompareAccountsReusedProps) => TAccountsDescription[];
+
+type TAccountTypesToFilter = (
+    | NonNullable<LandingCompany['mt_gaming_company']>['financial']
+    | NonNullable<LandingCompany['mt_financial_company']>['financial']
+    | NonNullable<LandingCompany['mt_financial_company']>['financial_stp']
+    | LandingCompany['dxtrade_gaming_company']
+    | LandingCompany['dxtrade_financial_company']
+    | boolean
+    | undefined
+)[];
+
+const getAccounts: TGetAccounts = ({ landing_companies, platform, is_logged_in, is_uk }) => {
     const getLoggedOutTypesCount = () => (platform === CFD_PLATFORMS.MT5 ? 3 : 2);
     const getLoggedInTypesCount = () =>
-        (platform === CFD_PLATFORMS.MT5
-            ? [
-                  landing_companies?.mt_gaming_company?.financial,
-                  landing_companies?.mt_financial_company?.financial,
-                  landing_companies?.mt_financial_company?.financial_stp && platform === CFD_PLATFORMS.MT5,
-              ]
-            : [landing_companies?.dxtrade_gaming_company, landing_companies?.dxtrade_financial_company]
+        (
+            (platform === CFD_PLATFORMS.MT5
+                ? [
+                      landing_companies?.mt_gaming_company?.financial,
+                      landing_companies?.mt_financial_company?.financial,
+                      landing_companies?.mt_financial_company?.financial_stp && platform === CFD_PLATFORMS.MT5,
+                  ]
+                : [
+                      landing_companies?.dxtrade_gaming_company,
+                      landing_companies?.dxtrade_financial_company,
+                  ]) as TAccountTypesToFilter
```

Comment:
I had to explicitly typify the array of objects because .filter() method in Typescript throws an error ("This expression is not callable") when an array that is being filtered contains objects of different types.